# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/RTCRtpReceiver.cpp
# Commit: c366748bd0bf
# Full Hash: c366748bd0bfebdfb48fcf83c648c261142edfd0
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 2
# Description:
#   Bug 1654112 - Get contributing/synchronization sources working again. r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106231
# ==============================================================================

diff -r d0b311007c03 -r c366748bd0bf dom/media/webrtc/jsapi/RTCRtpReceiver.cpp
--- a/dom/media/webrtc/jsapi/RTCRtpReceiver.cpp	Fri Feb 19 15:56:41 2021 -0600
+++ b/dom/media/webrtc/jsapi/RTCRtpReceiver.cpp	Tue Feb 23 18:13:34 2021 -0600
@@ -470,10 +470,8 @@
     nsTArray<RTCRtpContributingSource>& aSources) {
   // Duplicate code...
   if (mPipeline && mPipeline->mConduit) {
-    RefPtr<AudioSessionConduit> conduit(
-        *mPipeline->mConduit->AsAudioSessionConduit());
     nsTArray<dom::RTCRtpSourceEntry> sources;
-    conduit->GetRtpSources(sources);
+    mPipeline->mConduit->GetRtpSources(sources);
     sources.RemoveElementsBy([](const dom::RTCRtpSourceEntry& aEntry) {
       return aEntry.mSourceType != dom::RTCRtpSourceEntryType::Contributing;
     });
@@ -486,10 +484,8 @@
     nsTArray<dom::RTCRtpSynchronizationSource>& aSources) {
   // Duplicate code...
   if (mPipeline && mPipeline->mConduit) {
-    RefPtr<AudioSessionConduit> conduit(
-        *mPipeline->mConduit->AsAudioSessionConduit());
     nsTArray<dom::RTCRtpSourceEntry> sources;
-    conduit->GetRtpSources(sources);
+    mPipeline->mConduit->GetRtpSources(sources);
     sources.RemoveElementsBy([](const dom::RTCRtpSourceEntry& aEntry) {
       return aEntry.mSourceType != dom::RTCRtpSourceEntryType::Synchronization;
     });